### PR TITLE
menu model router and test

### PR DESCRIPTION
**Pull request by jmpaik**

### CODE REVIEW

File: route/menu-router.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    ACL: 'public-read',
+    Bucket: process.env.AWS_BUCKET,
+    Key: `${req.file.filename}${ext}`,
+    Body: fs.createReadStream(req.file.path)
+  };
+
+  Biz.findById(req.params.bizId)
+  .then( () => s3uploadProm(params))
+  .then( s3data => {
+    del([`${dataDir}/*`]);
```
Comment (geoffsimons) [inline route/menu-router.js]: This should be: del(req.file.path);

File: route/pic-router.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-
-const Router = require('express').Router;
-
-const picRouter = module.exports = Router();
-
-picRouter.post('/api/pic', function(req, res, next) {
-  //TODO: Implement POST /api/biz
-  next();
-});
+// 'use strict';
```
Comment (geoffsimons) [inline route/pic-router.js]: Just remove this file.

File: test/menu-route-test.js [hunk 3]
```diff
@@ -0,0 +1 @@
+'use strict';
```
Comment (geoffsimons) [inline test/menu-route-test.js]: I would let Kyler add this file. No need for you to upload the first version with only 'use strict';

File: test/pic-route-test.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
-'use strict';
-
-require('./lib/test-env.js');
+// 'use strict';
```
Comment (geoffsimons) [inline test/pic-route-test.js]: Remove this file for now

**Review by geoffsimons — COMMENTED** [review]
I think after the 4 changes proposed, this should be mergeable.

File: route/menu-router.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    let menuData = {
+      menuId: req.menu._id,
```
Comment (bazacko) [inline route/menu-router.js]: A user would not supply a menuId when creating a brand new menu - it's not in the schema anyway

File: model/menu.js [hunk 6]
```diff
@@ -1,13 +1,14 @@
-//Kyler Dotson
 'use strict';
 
 const Mongoose = require('mongoose');
+const Schema = mongoose.Schema;
```
Comment (bazacko) [inline model/menu.js:4]: syntax: capitalize mongoose

File: route/menu-router.js (hunk 5 above)
Comment (geoffsimons) [reply route/menu-router.js]: This looks like legacy from his prior model.

File: model/menu.js (hunk 6 above)
Comment (geoffsimons) [reply model/menu.js:4]: Alternatively, make the first mongoose lowercase (which is what is MUCH more common).

File: route/menu-router.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+  let params = {
+    ACL: 'public-read',
+    Bucket: process.env.AWS_BUCKET,
+    Key: `${req.file.filename}${ext}`,
+    Body: fs.createReadStream(req.file.path)
+  };
+
+  Biz.findById(req.params.bizId)
+  .then( () => s3uploadProm(params))
```
Comment (bazacko) [inline route/menu-router.js]: You found a Biz, but didn't do anything with it. We need to update the Biz with its new MenuID.

Comment (geoffsimons) [reply route/menu-router.js]: He sets bizId, and it probably should be businessId. See line 59

**Review by geoffsimons — COMMENTED** [review]
I think it's good to go after the last change.

File: route/menu-router.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  .then( s3data => {
+    del(req.file.path);
+    let menuData = {
+      
+      objectKey: s3data.Key,
+      imageURI: s3data.Location,
+      bizId: req.params.bizId
+    }
```
Comment (bazacko) [inline route/menu-router.js:56]: The properties used to create a new Menu don't correspond to the schema

File: route/menu-router.js (hunk 7 above)
Comment (bazacko) [reply route/menu-router.js]: I suppose it's somewhat redundant to put the Biz ID in the menu, and also put the menu ID in the biz.

**Pull request merged**